pygoodwe/__init__.py

```diff
@@ -125,6 +125,31 @@ def getCurrentReadings(self, raw=True, retry=1, maxretries=5, delay=30): #pylint
     #                 'eday_kwh': round(eday_kwh, 3)
     #             })
     #     return result
+    
+    def getDayDetailedReadingsExcel(self, date):
+        """ retrieves the detailed daily results of the given date as an Excel sheet
+            processing the Excel sheet is outside the scope of the current module """
+        datestr = datetime.strftime(date, "%Y-%m-%d")
+        payload = {
+            'date' : datestr,
+            'pw_id' : self.system_id,
+            'img_width': 1050,
+            'img_height': 268
+        }
+        self.data = self.call("v1/PowerStation/ExportPowerstationPac", payload)
+        
+        if self.data is not None:
+            payload = {
+                'id' : self.data,
+            }
+            self.data = self.call("v1/ReportData/GetStationPowerDataFilePath", payload)
+        
+        if self.data is not None and self.data["file_path"] is not None:
+            r = requests.get(self.data["file_path"], allow_redirects=True)
+            open('Plant_Power_%s.xls' % datestr, 'wb').write(r.content)
```

Comment:
This needs to be settable in the function input, I don't feel comfortable having something that may/will write to unexpected places on disk. Maybe add `**kwargs` to the function def similar to the `img_width` and `img_height` options above and use this?

```python
output_filename = kwargs.get(filename, f"'Plant_Power_{datestr}.xls")
open(output_filename, 'wb').write(r.content)
```